protocol/contracts/oracle/src/state/config.rs

```diff
@@ -35,4 +36,32 @@ impl Config {
             .map(mem::drop)
             .map_err(ContractError::UpdateConfig)
     }
+
+    pub fn migrate(storage: &mut dyn Storage) -> ContractResult<()> {
+        use currency::SymbolOwned;
+        use serde::{Deserialize, Serialize, Serializer};
+
+        const STORAGE_V0: Item<'static, OldConfig> = Item::new("config");
+
+        #[derive(Deserialize)]
+        #[serde(deny_unknown_fields, rename_all = "snake_case")]
+        struct OldConfig {
+            #[allow(dead_code)]
+            base_asset: SymbolOwned,
+            price_config: PriceConfig,
+        }
+        impl Serialize for OldConfig {
+            fn serialize<S>(&self, _serializer: S) -> Result<S::Ok, S::Error>
+            where
+                S: Serializer,
+            {
+                unreachable!("satisfy Item::load requirements")
```

Comment:
[Note] For reference, the documentation of [`Result::expect`](https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.expect) \(in the end\) hints towards saying what it is expected to be.  
In this case, perhaps leaving it as a comment on the `impl` explaining why it was done might be better, while having the `unreachable!` macro be replaced with `unimplemented!` explaining the type shouldn't be serialized.

While essentially both `unimplemented!` and `unreachable!` are just a `panic!` underneath, they do imply different things to the reader.

Reply:
maybe I am missing something, but what is the relation between `expect` and `unimplemented!` except they panic under specific circumstances

Reply:
Well, both signify state that is for one way or another unsatisfied or unwanted.  
I referred to it as an example because it gave hints on better practices for error messages. That's all.